core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -683,3 +729,56 @@ trait TemporalSchema {
     )(formatter)
 
 }
+
+case class FieldType(isInput: Boolean, isSubscription: Boolean)
+
+case class PartiallyAppliedField[V](
+  name: String,
+  description: Option[String],
+  directives: List[Directive]
+) {
+  def apply[R, V1](fn: V => V1)(implicit ev: Schema[R, V1], ft: FieldType): (__Field, V => Step[R]) =
```

Comment:
How about a variant that could be used for `eitherSchema`? It could take a `V => Either[Step[R], V1`. Is that too specific?

Reply:
I'll see if that would work, my only concern would be that it would confuse the inference in some cases.

Reply:
If the inference is degraded, maybe we name the other something else than `apply`?